src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -561,8 +565,35 @@ scaleNominalToMergeCredit (NominalDebt             nominalDebt)
     let mergeCredits_spec = floor $ toRational nominalCredits
                                   * toRational mergeDebt
                                   / toRational nominalDebt
+        mergeCredits_fast = w2i $ timesDivABC_fast (i2w nominalCredits)
+                                                   (i2w mergeDebt)
+                                                   (i2w nominalDebt)
```

Comment:
Okay so we are using floor in the spec because the fast impl also is flooring

Reply:
Yes exactly. And ultimately it doesn't matter since once we hit 100% then we get the right answer exactly.

So we can always be 1 behind and that's ok, because the rounding errors do not compound, since we convert absolute nominal to absolute merging credit.

Reply:
I've added a comment to this effect.
```
        -- Note that it doesn't matter if we use floor or ceiling here.
        -- Rounding errors will not compound because we sum nominal debt and
        -- convert absolute nominal to absolute merging credit. We don't
        -- convert each deposit and sum all the rounding errors.
        -- When nominalCredits == nominalDebt then the result is exact anyway
        -- (being mergeDebt) so the rounding mode makes no difference when we
        -- get to the end of the merge. Using floor makes things simpler for
        -- the fast integer implementation below, so we take that as the spec.
```